tensorflow_privacy/privacy/membership_inference_attack/data_structures.py

```diff
@@ -529,6 +529,39 @@ def __str__(self):
     ])
 
 
+@dataclass
+class SingleRiskScoreResult:
+  """Results from computing privacy risk scores.
+  this part is quite preliminary: it shows how to leverage privacy risk score to perform attacks with thresholding on risk score
+  """
+
+  # Data slice this result was calculated for.
+  slice_spec: SingleSliceSpec
+
+  train_risk_scores: np.ndarray 
+  
+  test_risk_scores: np.ndarray
+  
+  def attack_with_varied_thresholds(self, threshold_list):
+    precision_list = []
+    recall_list = []
+    meaningful_threshold_list = []
+    for threshold in threshold_list:
+        true_positive_normalized = np.sum(self.train_risk_scores>=threshold)/(len(self.train_risk_scores)+0.0)
+        false_positive_normalized = np.sum(self.test_risk_scores>=threshold)/(len(self.test_risk_scores)+0.0)
+        if true_positive_normalized+false_positive_normalized>0:
+          meaningful_threshold_list.append(threshold)
+          precision_list.append(true_positive_normalized/(true_positive_normalized+false_positive_normalized+0.0))
+          recall_list.append(true_positive_normalized)
+    return meaningful_threshold_list, precision_list, recall_list
+  
+  def print_results(self, threshold_list=[1,0.9,0.8,0.7,0.6,0.5]):
```

Comment:
Why is the default value this threshold list? Can you add a comment and a test for this function?

Reply:
I added a comment to it. Will further edit this function later

Reply:
Just rewrote this function!